sdk/src/assertions/actions.rs

```diff
@@ -486,6 +490,10 @@ pub struct Actions {
     /// A list of [`Action`]s.
     pub actions: Vec<Action>,
 
+    /// A list of of the software/hardware that did the action.
+    #[serde(rename = "softwareAgents", skip_serializing_if = "Option::is_none")]
+    pub software_agents: Option<Vec<ClaimGeneratorInfo>>,
```

Comment:
If you are adding sofwareAgents we should also add sofwareAgentIndex to action, the two require each other.

Reply:
The index is in the Action

Reply:
yes, it sure is, and I guess I added it :)